pkgs/_test_macros/lib/json_codable.dart

```diff
@@ -49,7 +49,28 @@ class JsonCodableImplementation implements Macro {
     final target = request.target;
     final model = await host.query(Query(target: target));
     final clazz = model.uris[target.uri]!.scopes[target.name]!;
-    // TODO(davidmorgan): check for super `fromJson`.
+
+    MacroScope.current.addModel(model);
```

Comment:
Fwiw, just reading this code I don't have any idea what it does, I have to go read the API docs which isn't great.